include/crocoddyl/core/activations/quadratic-barrier.hpp

```diff
@@ -40,19 +40,25 @@ struct ActivationBoundsTpl {
       throw_pretty("Invalid argument: "
                    << "The range of beta is between 0 and 1");
     }
-    using std::isfinite;
     for (std::size_t i = 0; i < static_cast<std::size_t>(lb.size()); ++i) {
       if (isfinite(lb(i)) && isfinite(ub(i))) {
         if (lb(i) - ub(i) > 0) {
           throw_pretty("Invalid argument: "
                        << "The lower and upper bounds are badly defined; ub has to be bigger / equals to lb");
         }
       }
+      // Assign the maximum value for infinity/nan values
+      if (!std::isfinite(lb(i))) {
```

Comment:
Anyways, I remember that there was an issue with the definition of isfinite for codegen classes. As a result, I had fixed the implementation here: https://github.com/stack-of-tasks/pinocchio/blob/master/src/codegen/cppadcg.hpp#L67. 
This was to let the compiler choose which `isfinite` to use depending on the scalar type.
So It would be better to not specify `std::isfinite`, but rather use `isfinite` instead.

Reply:
done